### PR TITLE
Fix deprecation Warnings for julia 0.7

### PR DESCRIPTION
**Pull request by felixcremer**
I used the workflow as explained here:
https://discourse.julialang.org/t/ann-introducing-upgradathon-fridays/12047/3
to get rid of the deprecation warnings in 0.7. 
The tests are running through locally on 0.7. 
This drops support for 0.6.

### CODE REVIEW

**Comment by visr** [comment]
Awesome, thanks! Could you also set up Travis and AppVeyor config files to stop testing 0.6, and add 0.7 (next to nightly)?

File: appveyor.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 environment:
   matrix:
-  - JULIA_URL: "https://julialang-s3.julialang.org/bin/winnt/x86/0.6/julia-0.6-latest-win32.exe"
-  - JULIA_URL: "https://julialang-s3.julialang.org/bin/winnt/x64/0.6/julia-0.6-latest-win64.exe"
+  - JULIA_URL: "https://julialang-s3.julialang.org/bin/winnt/x86/0.6/julia-0.7-latest-win32.exe"
+  - JULIA_URL: "https://julialang-s3.julialang.org/bin/winnt/x64/0.6/julia-0.7-latest-win64.exe"
```
Comment (visr) [inline appveyor.yml]: Ah there is another /0.6/ in there which should be /0.7/

File: src/geos_operations.jl [hunk 2]
```diff
@@ -308,7 +308,7 @@ isClosed(obj::LineString) = isClosed(obj.ptr) # Call only on LINESTRING
 
 # Converts Geometry to normal form (or canonical form).
 for geom in (:Point, :MultiPoint, :LineString, :MultiLineString, :LinearRing, :Polygon, :MultiPolygon, :GeometryCollection)
-    @eval Base.normalize!(obj::$geom) = normalize!(obj.ptr)
+    @eval LinearAlgebra.normalize!(obj::$geom) = normalize!(obj.ptr)
```
Comment (yeesian) [inline src/geos_operations.jl]: This should now be `@eval normalize!(obj::$geom) = normalize!(obj.ptr)`

We shouldn't need the dependency on LinearAlgebra.

File: src/geos_functions.jl [hunk 3]
```diff
@@ -941,7 +941,7 @@ getGeometries(ptr::GEOSGeom, context::GEOSContext = _context) =
 
 # Converts Geometry to normal form (or canonical form).
 # Return -1 on exception, 0 otherwise.
-function Base.normalize!(ptr::GEOSGeom, context::GEOSContext = _context)
+function LinearAlgebra.normalize!(ptr::GEOSGeom, context::GEOSContext = _context)
```
Comment (yeesian) [inline src/geos_functions.jl]: This should now be `function normalize!(ptr::GEOSGeom, context::GEOSContext = _context)`

We shouldn't need the dependency on LinearAlgebra.

**Comment by yeesian** [comment]
The tests need to be updated too: from `normalize!(...)` to `LibGEOS.normalize!(...)` in
- test_geos_operations ([lines 148-156](https://github.com/JuliaGeo/LibGEOS.jl/blob/f7bcf02b2ebaf9d43bf4f4cc5cb3b96601ddfd11/test/test_geos_operations.jl#L148-L156))
- test_geos_functions ([lines 343-361](https://github.com/JuliaGeo/LibGEOS.jl/blob/f7bcf02b2ebaf9d43bf4f4cc5cb3b96601ddfd11/test/test_geos_functions.jl#L343-L361))

**Comment by yeesian** [comment]
Thanks!

**Pull request merged**